snoop/utils.py

```diff
@@ -123,3 +132,39 @@ def wrapper(*args, **kwargs):
     decorator.cm = cm
 
     return decorator
+
+@contextmanager
+def flock(fd):
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX)
+        yield fd
+    finally:
+        fcntl.flock(fd, fcntl.LOCK_UN)
+
+def append_with_lock(path, bytes):
```

Comment:
There's really no need for locking. If you open the file in "append" mode, writes are atomic, as long as they fit in the (kernel's?) write buffer, and our log messages are short enough, for the most part. If, by some obscure reason a couple of messages get garbled, we can fix them by hand, or just skip them. I'm slightly concerned about the runtime overhead involved in locking, and more-than-slightly concerned about all this extra code :)